src/model/add-dots.ts

```diff
@@ -0,0 +1,60 @@
+// Check if the block contains a dot, replace with an empty span with a bullet attribute
+const hasDots = (element: CAPIElement): boolean => {
+	if (element._type !== 'model.dotcomrendering.pageElements.TextBlockElement')
+		return false;
+
+	if (element.html.includes('•')) {
+		element.html = element.html.replace(
+			new RegExp('[•]', 'g'),
+			'<span data-dcr-style="bullet"></span>',
+		);
+		return true;
+	}
+
+	return false;
+};
+
+const checkForDots = (elements: CAPIElement[]): CAPIElement[] => {
+	// Loop over elements and check if a dot is in the TextBlockElement
+	const enhanced: CAPIElement[] = [];
+	elements.forEach((element) => {
+		if (
+			element._type ===
+				'model.dotcomrendering.pageElements.TextBlockElement' &&
+			hasDots(element)
+		) {
+			enhanced.push({
+				...element,
+			});
+		} else {
+			enhanced.push(element);
+		}
```

Comment:
I think this is mutating the original element, which is something we generally want to avoid. The hasDots function should return a boolean and then a decision branched from there.

Eg. We can use the spread pattern to avoid mutation:

return {
  ...originalThing,
  someThing: newValue,
}

See: https://css-tricks.com/understanding-immutability-in-javascript/

Immutability is esp. useful in js because objects are passed as references, which leads to confusing errors.

Note. Some codebases will 'freeze' objects to make mutation illegal. We don't do this but we want out code to allow this without needing to be refactored.